### PR TITLE
Decrease access level to read-only

### LINKED ISSUE #15638: 🐛 Bug Report: Kubernetes plugin asks to login even if already logged in using google auth
### 📜 Description

Trying to see kubernetes plugins page, after logging in using google auth provider.

### 👍 Expected behavior

Kubernetes plugin should not ask to reauthenticate as user is already authenticated with google.

### 👎 Actual Behavior with Screenshots

Kubernetes plugin asks to reauthenticate.

image

### 👟 Reproduction steps

- Login using google auth
- Go to kubernetes plugin page
- A pop up is thrown to login using Google

### 🖥️ Your Environment

**Configuration**:
```
kubernetes:
  serviceLocatorMethod:
    type: 'multiTenant'
  clusterLocatorMethods:
    - type: 'gke'
      projectId: 'my-project-id'
      region: 'australia-southeast1'
      skipTLSVerify: true
  customResources:
    - group: 'argoproj.io'
      apiVersion: 'v1alpha1'
      plural: 'rollouts'
```

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by avkudryashov**
## Hey, I just made a Pull Request!

Changed scope to read-only for kubernetes auth-provider

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

close #15638

### CODE REVIEW

**Comment by benjdlambert** [comment]
@backstage/kubernetes-maintainers ping!

**Comment by benjdlambert** [comment]
@jamieklassen @mclarke47 does this seem sane?

**Review by jamieklassen — APPROVED** [review]
I think this is sane, and a best practice, but when I tested it I still had to re-authenticate to get the `cloud-platform.read-only` scope so I disagree that it resolves #15638.

**Comment by avkudryashov** [comment]
> I think this is sane, and a best practice, but when I tested it I still had to re-authenticate to get the `cloud-platform.read-only` scope so I disagree that it resolves #15638.

Should I create another issue to reduce auth scope for the Kubernetes plugin?

File: plugins/kubernetes/src/kubernetes-auth-provider/GoogleKubernetesAuthProvider.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export class GoogleKubernetesAuthProvider implements KubernetesAuthProvider {
   async getCredentials(): Promise<{ token: string }> {
     return {
       token: await this.authProvider.getAccessToken(
-        'https://www.googleapis.com/auth/cloud-platform',
```
Comment (mclarke47) [inline plugins/kubernetes/src/kubernetes-auth-provider/GoogleKubernetesAuthProvider.ts:42]: I wonder would this prevent us from doing patches/deletes against objects through the proxy API when using this auth @jamieklassen

Comment (jamieklassen) [reply plugins/kubernetes/src/kubernetes-auth-provider/GoogleKubernetesAuthProvider.ts:42]: lol of course it would yep

Comment (avkudryashov) [reply plugins/kubernetes/src/kubernetes-auth-provider/GoogleKubernetesAuthProvider.ts:42]: Should we have `read-only` auth scope value as the default?
 Will the configuration should look like this one?:
 ```yaml
  clusterLocatorMethods:
    - type: 'gke'
      projectId: 'my-project'
      authProvider: google
      authScopes: []
 ```

Comment (jamieklassen) [reply plugins/kubernetes/src/kubernetes-auth-provider/GoogleKubernetesAuthProvider.ts:42]: Actually Google's authN model might not work the way I thought. When I checked out this code, visited the Kubernetes tab, opened my dev tools and copied the access token used to get objects for an entity, I was able to use the same token to create a namespace.

Comment (mclarke47) [reply plugins/kubernetes/src/kubernetes-auth-provider/GoogleKubernetesAuthProvider.ts:42]: Sure we can merge, if we discover issues we can raise a new bug

**Pull request merged**